fvm/meter/interaction_meter.go

```diff
@@ -117,14 +132,31 @@ func GetStorageKeyValueSizeForTesting(
 	return getStorageKeyValueSize(storageKey, value)
 }
 
-func (m *InteractionMeter) GetStorageUpdateSizeMapForTesting() MeteredStorageInteractionMap {
-	return m.storageUpdateSizeMap
+func (m *InteractionMeter) GetStorageRWSizeMapForTesting() (
+	reads MeteredStorageInteractionMap,
+	writes MeteredStorageInteractionMap,
+) {
+	return m.reads, m.writes
 }
 
+// Merge merges the child interaction meter into the parent interaction meter
+// Prioritise parent reads because they happened first
+// Prioritise child writes because they happened last
 func (m *InteractionMeter) Merge(child InteractionMeter) {
-	for key, value := range child.storageUpdateSizeMap {
-		m.storageUpdateSizeMap[key] = value
+	for key, value := range child.reads {
+		_, parentRead := m.reads[key]
+		if parentRead {
+			continue
+		}
+
+		m.reads[key] = value
+		m.totalStorageBytesRead += value
+	}
+
+	for key, value := range child.writes {
+		parentWriteBytes := m.writes[key]
+		m.totalStorageBytesWritten -= parentWriteBytes
+		m.writes[key] = value
+		m.totalStorageBytesWritten += value
```

Comment:
we do `-=` before `+=`, so the first `-=` would cause underflow. 

can we change `totalStorageBytesWritten` into `int64`?